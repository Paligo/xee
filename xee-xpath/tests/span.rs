use insta::assert_debug_snapshot;

use xee_xpath::run_without_context;

#[test]
fn test_right_left_side() {
    let expr = "0 + 12 + ((2, 3, 4) + 1)";
    //          012345678901234567890123
    //                    10           23
    //  So from 10, 13 is expected
    // we can't atomize the sequence which happens because of addition
    assert_debug_snapshot!(run_without_context(expr));
}

#[test]
fn test_right_right_side() {
    let expr = "0 + 12 + (1 + (2, 3, 4))";
    //          012345678901234567890123
    //                    10           23
    //  So from 10, 13 is expected
    // we can't atomize the sequence which happens because of addition
    assert_debug_snapshot!(run_without_context(expr));
}
