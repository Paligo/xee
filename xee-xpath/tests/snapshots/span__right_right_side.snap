---
source: xee-xpath/tests/span.rs
expression: run_without_context(expr)
---
Err(
    XPTY0004 {
        src: NamedSource {
            name: "input",
            source: "<redacted>",
        ,
        span: SourceSpan {
            offset: SourceOffset(
                10,
            ),
            length: SourceOffset(
                13,
            ),
        },
    },
)
